src/app/shared/models/student-grades.spec.ts

```diff
@@ -111,8 +111,20 @@ describe('student-grade utils', () => {
       results[1].grades.every((grade) => grade.kind === 'no-result')
     ).toBeTruthy();
 
-    expect(results[0].finalGrade).toBeUndefined;
-    expect(results[1].finalGrade).toBeUndefined;
+    expect(results[0].finalGrade).toEqual({
+      id: undefined,
+      average: null,
+      finalGradeId: undefined,
+      freeHandGrade: NaN,
+      canGrade: false,
+    });
+    expect(results[1].finalGrade).toEqual({
+      id: undefined,
+      average: null,
+      finalGradeId: undefined,
+      freeHandGrade: NaN,
+      canGrade: false,
+    });
```

Comment:
@caebr Die Expectation wurde hier gar nicht ausgeführt (`toBeUndefined` ohne Aufruf). Nun kommt da in diesem Fall aber ein Objekt daher. Und insb. der `NaN` Wert ist nicht sehr schön (Division durch 0?).

Sehen die Werte für dich grundsätzlich so aus wie wir sie erwarten würden? Und weisst du gerade wo wir das `NaN` fixen müssen?

Reply:
Also  `course.Gradings = [];` wäre eher `null` vom Backend. 

Vielleicht können wir das heute kurz zusammen anschauen.